src/libslic3r/Brim.cpp

```diff
@@ -897,8 +901,52 @@ static ExPolygons outer_inner_brim_area(const Print& print,
                         if (brim_type == BrimType::btOuterOnly || brim_type == BrimType::btOuterAndInner || brim_type == BrimType::btAutoBrim) {
                             // BBS: inner and outer boundary are offset from the same polygon incase of round off error.
                             auto innerExpoly = offset_ex(ex_poly.contour, brim_offset, jtRound, SCALED_RESOLUTION);
-                            append(brim_area_object, diff_ex(offset_ex(innerExpoly, brim_width_mod, jtRound, SCALED_RESOLUTION), innerExpoly));
 
+                            if (use_brim_ears) {
+                                // Generate ears. Ported from SuperSlicer: https://github.com/supermerill/SuperSlicer/blob/45d0532845b63cd5cefe7de7dc4ef0e0ed7e030a/src/libslic3r/Brim.cpp#L1116
+                                coord_t size_ear = (brim_width_mod - brim_offset - flow.scaled_spacing());
+                                if (size_ear > 0) {
```

Comment:
While it's good to have a sanity check here, I believe the root cause if because ear use brim_width too.
When brim_width is 0, we should disable ear.
For Orca, it makes more sense to add mouse ear as a new brim type?
<img width="417" alt="Screenshot 2023-08-13 at 1 47 13 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/103989404/e0ce74fe-c396-4c08-9e4c-904d1e84ff5f">

Reply:
Then do we want having brim ears with/without inner brim?

Reply:
I believe mouse ears is exclusive with other brim type(this is the also the behavior in SS).
What do you think?

Reply:
When user chose mouse ears, we won't generate any other type of brims

Reply:
This is how it works in SS:
You could have both brim ears and inner brim:
![image](https://github.com/SoftFever/OrcaSlicer/assets/1537155/668990ad-0c41-4bf5-ad9a-a4111abfd9fc)
![image](https://github.com/SoftFever/OrcaSlicer/assets/1537155/56bb598e-a42b-457f-b89b-83c1436a618a)

The brim ear option is disabled unless outer brim is enabled:
![image](https://github.com/SoftFever/OrcaSlicer/assets/1537155/0c71a44e-388e-401a-b134-12fe3fc909be)

Reply:
> This is how it works in SS: You could have both brim ears and inner brim: ![image](https://user-images.githubusercontent.com/1537155/260283972-668990ad-0c41-4bf5-ad9a-a4111abfd9fc.png) ![image](https://user-images.githubusercontent.com/1537155/260283980-56bb598e-a42b-457f-b89b-83c1436a618a.png)
> 
> The brim ear option is disabled unless outer brim is enabled: ![image](https://user-images.githubusercontent.com/1537155/260284009-0c71a44e-388e-401a-b134-12fe3fc909be.png)

It makes more sense to me to automatically create ears wherever the corner qualifies, no matter it's inner or outer.
Thoughts?

Reply:
Do you mean when user select brim ear type, then it adds ears to both inner and outer corners that qualify?

Reply:
Yes.

Reply:
Make sense. I'll give it a try.

Reply:
What about support brims btw?